src/repl.js

```diff
@@ -96,21 +97,35 @@ Prototype REPL - https://github.com/nodejs/repl`,
     const evaluateResult = await this.eval(line, awaited);
 
     if (evaluateResult.exceptionDetails) {
+      // lets try for recovering
+      const result = isRecoverableError(evaluateResult.exceptionDetails.exception, line);
+      if (result) {
+        this.io.setPrefix('...');
+        this.io.setMultiline(true);
```

Comment:
`setMultiline` should handle the `...` to keep the prefix consistent (see screenshot) also we have to discard the return value of `onLine` because it dumps `undefined` when a new multiline happens (see screenshot)

![](https://gc.gy/ʦ⟞菉.png)

Reply:
something a bit more "flowing" might be to return some symbol `IO.kNeedsAnotherLine` from `onLine` to tell the IO to buffer another line

Reply:
@devsnek Sure, will work on these things. I don't get your last statement though:

>something a bit more "flowing" might be to return some symbol (IO.kNeedsAnotherLine or something) from onLine to tell the IO to buffer another line